core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/parser/YamlSignerParser.java

```diff
@@ -12,35 +12,52 @@
  */
 package tech.pegasys.eth2signer.core.multikey.metadata.parser;
 
-import tech.pegasys.eth2signer.core.multikey.metadata.ArtifactSignerFactory;
+import static java.util.Collections.singletonList;
+
+import tech.pegasys.eth2signer.core.multikey.metadata.BlsArtifactSignerFactory;
+import tech.pegasys.eth2signer.core.multikey.metadata.Secp256k1ArtifactSignerFactory;
 import tech.pegasys.eth2signer.core.multikey.metadata.SigningMetadata;
 import tech.pegasys.eth2signer.core.multikey.metadata.SigningMetadataException;
 import tech.pegasys.eth2signer.core.signing.ArtifactSigner;
+import tech.pegasys.eth2signer.core.signing.KeyType;
 
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.List;
 
 import com.fasterxml.jackson.core.JsonParseException;
 import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 
 public class YamlSignerParser implements SignerParser {
+
   private static final ObjectMapper OBJECT_MAPPER =
       new ObjectMapper(new YAMLFactory()).registerModule(new SigningMetadataModule());
-  final ArtifactSignerFactory artifactSignerFactory;
 
-  public YamlSignerParser(final ArtifactSignerFactory artifactSignerFactory) {
-    this.artifactSignerFactory = artifactSignerFactory;
+  private final BlsArtifactSignerFactory blsFactory;
+  private final Secp256k1ArtifactSignerFactory ethSecpFactory;
+  private final Secp256k1ArtifactSignerFactory fcSecpFactory;
+
+  public YamlSignerParser(
+      final BlsArtifactSignerFactory blsFactory,
```

Comment:
This doesn't seem quite right. I wouldn't expect the YamlSignerParser to have deal with determining which factory to use for a keyType. Thinking that perhaps the keyType should be a property of the ArtifactSignerFactory then could just pass in a List<ArtifactSignerFactory> and then do a transform of List<ArtifactSignerFactory>.filter(metaDataInfo.getKeyType()).

Reply:
yeah that's nicer.